pytest_plugins/fixture_markers.py

```diff
@@ -39,7 +39,7 @@ def pytest_generate_tests(metafunc):
             metafunc.parametrize(
                 'rhel_contenthost',
                 rhel_params,
-                ids=[f'rhel{r["rhel_version"]}' for r in rhel_params],
+                ids=sorted(f'rhel{r["rhel_version"]}' for r in rhel_params),
```

Comment:
I think this leaves us open for rhel_params and the ids list to end up out of order with each other.

The params list should be sorted, and then the ids picked from the sorted list, so that when the parameters and their IDs are zipped together they don't get out of order.

```
In [31]: rhel_params = [{'rhel_version': '7', 'rhel': '7.6'}, {'rhel_version': '8', 'rhel': '8.5'}, {'rhel_version': '7_fips', 'rhel': '7 fips'}, {'rhel_version': '8_fips', 'rhel': '8 fips'}]

In [32]: for i in zip(rhel_params, sorted([f"rhel{r['rhel_version']}" for r in rhel_params])):
    ...:     print(i)
    ...: 
({'rhel_version': '7', 'rhel': '7.6'}, 'rhel7')
({'rhel_version': '8', 'rhel': '8.5'}, 'rhel7_fips')
({'rhel_version': '7_fips', 'rhel': '7 fips'}, 'rhel8')
({'rhel_version': '8_fips', 'rhel': '8 fips'}, 'rhel8_fips')
```

Versus if we sort the param list first:

```
In [31]: rhel_params = [{'rhel_version': '7', 'rhel': '7.6'}, {'rhel_version': '8', 'rhel': '8.5'}, {'rhel_version': '7_fips', 'rhel': '7 fips'}, {'rhel_version': '8_fips', 'rhel': '8 fips'}]

In [32]: for i in zip(rhel_params, sorted([f"rhel{r['rhel_version']}" for r in rhel_params])):
    ...:     print(i)
    ...: 
({'rhel_version': '7', 'rhel': '7.6'}, 'rhel7')
({'rhel_version': '8', 'rhel': '8.5'}, 'rhel7_fips')
({'rhel_version': '7_fips', 'rhel': '7 fips'}, 'rhel8')
({'rhel_version': '8_fips', 'rhel': '8 fips'}, 'rhel8_fips')

```

Maybe its still possible for xdist workers to collect in different orders, in which case we could be extra sure and sort the ids list *again* - but in that case it will always be the same order as the parameter values are parametrized in.